src/server/controllers/incidents.js

```diff
@@ -158,7 +158,13 @@ module.exports = {
   },
 
   // get all incidents
-  list(req, res) {
+  async list(req, res) {
+    if (res.locals.roleId === 2) {
+      const userAssignedIncidents = await User.findOne({ 
+        where: { id: res.locals.id}, 
+        include: [{model: Incident, as:'assignedIncidents'}]});
+      return userAssignedIncidents;
```

Comment:
I think `userAssignedIncidents` also needs mapping to return proper results.
We can say:
```js
let query;
if (res.locals.roleId === 2) query = User.findOne({ 
        where: { id: res.locals.id}, 
        include: [{model: Incident, as:'assignedIncidents'}]});
else if (res.locals.roleId === 3) query = Incident.findAll({ include: includes });
return query.then(incidents => { /* Do the necessary mapping here */}
```